src/flamegraph/color.rs

```diff
@@ -203,25 +213,27 @@ fn rgb_components_for_palette(
     v2: f32,
     v3: f32,
 ) -> (u8, u8, u8) {
-    let real_palette = match palette {
-        Palette::Hot => return (t!(205, 50, v3), t!(0, 230, v1), t!(0, 55, v2)),
-        Palette::Mem => return (t!(0, 0, v3), t!(190, 50, v2), t!(0, 210, v1)),
-        Palette::Io => return (t!(80, 60, v1), t!(80, 60, v1), t!(190, 55, v2)),
-        Palette::Red => return (t!(200, 55, v1), t!(50, 80, v1), t!(50, 80, v1)),
-        Palette::Green => return (t!(50, 60, v1), t!(200, 55, v1), t!(50, 60, v1)),
-        Palette::Blue => return (t!(80, 60, v1), t!(80, 60, v1), t!(205, 50, v1)),
-        Palette::Yellow => return (t!(175, 55, v1), t!(175, 55, v1), t!(50, 20, v1)),
-        Palette::Purple => return (t!(190, 65, v1), t!(80, 60, v1), t!(190, 65, v1)),
-        Palette::Aqua => return (t!(50, 60, v1), t!(165, 55, v1), t!(165, 55, v1)),
-        Palette::Orange => return (t!(190, 65, v1), t!(90, 65, v1), t!(0, 0, v1)),
-        Palette::Java => handle_java_palette(name),
-        Palette::Perl => handle_perl_palette(name),
-        Palette::Js => handle_js_palette(name),
-        Palette::Wakeup => handle_wakeup_palette(name),
-        Palette::Chain => handle_chain_palette(name),
+    let basic_palette = match palette {
+        Palette::Basic(basic) => basic.to_owned(),
```

Comment:
If you `derive(Copy)` for `BasicPalette`, you can just make this `=> basic,`

Reply:
[Ok](https://github.com/jonhoo/inferno/pull/32/commits/40386d59b8986d0089d0c998c339915be353062a)